static/js/settings_account.js

```diff
@@ -140,8 +140,20 @@ exports.initialize_custom_date_type_fields = function (element_id) {
     $(element_id).find(".custom_user_field .datepicker").flatpickr({
         altInput: true,
         altFormat: "F j, Y"});
-};
 
+    $(element_id).find(".custom_user_field ").on("mouseenter", function () {
+        if ($(element_id).find(".custom_user_field .datepicker").val() !== '') {
+            $(element_id).find(".remove_date").show();
+        } else {
+            $(element_id).find(".remove_date").hide();
+        }
+
+    });
```

Comment:
These click handlers will reset every `datepicker` field in `$(element_id)`. Therefore, we need to only modify value of event field. You can test this case by adding multiple date type field in organization and try to change value of one of them in user settings.